src/main/scala/firrtl/Emitter.scala

```diff
@@ -879,6 +885,10 @@ class VerilogEmitter extends SeqTransform with Emitter {
           simulate(sx.clk, sx.en, stop(sx.ret), Some("STOP_COND"), sx.info)
         case sx: Print =>
           simulate(sx.clk, sx.en, printf(sx.string, sx.args), Some("PRINTF_COND"), sx.info)
+        // formal
+        case sx: Check =>
+          check(sx.arg)
```

Comment:
The enclosing match is used as a way to call side-effect-having functions -- the evaluated value of the match is not used.

The `Print` and `Stop` cases call the impure `simulate` to cause procedures containing the appropriate Verilog displays or finishes to be emitted. If `Check` had a clock, you could make some call resembling:
`simulate(sx.clk, Utils.one, check(sx.arg), ...`

Unfortunately, there doesn't seem to be a utility like `simulate` for emitting wildcard-sensitivity procedures, so you'd have to extend it to make the clock argument optional.

Your intuition to just return a value is certainly understandable given that there seems to be a lot of action-at-a-distance going on in this function :)

Reply:
Thanks Albert! I'm adding a `formals` array buffer to hold the formal checks, to be emitted at the end of the file within an `\`ifdef FORMAL` block.